File: src/main/java/org/cubewhy/celestial/event/impl/AuthEvent.java
/*
 * Celestial Launcher <me@lunarclient.top>
 * License under GPLv3
 * Do NOT remove this note if you want to copy this file.
 */

package org.cubewhy.celestial.event.impl;

import org.cubewhy.celestial.event.Event;

import java.net.URL;

public class AuthEvent extends Event {

    public final URL authURL;
    private volatile String result = null;

    public AuthEvent(URL authURL) {
        this.authURL = authURL;
    }

    public String waitForAuth() {
        while (result == null) {
            Thread.onSpinWait();
        }
        return result;
    }

    public void put(String url) {
        this.result = url;
    }
}

File: src/main/kotlin/org/cubewhy/celestial/event/impl/AuthEvent.kt
/*
 * Celestial Launcher <me@lunarclient.top>
 * License under GPLv3
 * Do NOT remove this note if you want to copy this file.
 */
package org.cubewhy.celestial.event.impl

import org.cubewhy.celestial.event.Event
import java.net.URL
import java.util.concurrent.atomic.AtomicBoolean

class AuthEvent(@JvmField val authURL: URL) : Event() {
    private var result = ""
    private val responded = AtomicBoolean(false)

    fun waitForAuth(): String {
        while (!responded.get()) {
            Thread.onSpinWait()
        }
        return result
    }

    fun put(url: String?) {
        if (url != null) {
            this.result = url
        }
        responded.set(true)
    }
}
